### DISCUSSION TITLE
How to allow dashes in select match key?

### DISCUSSION THREAD

**Discussion by mdrayer** [amannn/next-intl#965] [Q&A] [answered]
For the `select` argument, I am having trouble setting up the match keys where one has a dash. Example:
```
{locale, select, es {Español} en-US {English} other {Unknown}}
```
Here, the "keys" are `es`, `en-US`, and `other`. I get the following error:
```
'EXPECT_SELECT_ARGUMENT_SELECTOR_FRAGMENT ({locale, select, es {Español} en-US {English} other {Unknown}})'
```
I'm not seeing any specifics on what is a valid key in the docs (https://next-intl-docs.vercel.app/docs/usage/messages#selecting-enum-based-values). Am I to assume that dashes aren't allowed, or is there some syntax magic I need to be using here? I have tried various syntax (`'en-US'`, `"en-US"`, `en'-US`, `[en-US]`, `(en-US)`), but none seem to work.

**Comment by amannn** [comment, marked as answer]
Thanks for the great question! I've clarified this in the docs in https://github.com/amannn/next-intl/pull/967, hope this helps!

**Reply by mdrayer** [reply]
Thanks, it does! Only alphanumeric and underscores allowed 👍

**Comment by cueedee** [comment]
Maybe I'm connecting dots where I shouldn't, but doesn't the ICU spec say that dashes should be allowed here?

At least that is how I interpret: https://icu.unicode.org/design/formatting/select : (note the final `-` in the `[a-zA-Z0-9_-]` regexp character class)

> - The pattern string for SelectFormat allows user-defined keywords that match the pattern `[a-zA-Z][a-zA-Z0-9_-]*` as well as the predefined keyword "`other`". ...
